.github/workflows/unittest.yaml

```diff
@@ -3,14 +3,34 @@ on:
   pull_request:
 
 jobs:
+  changes:
+    runs-on: ubuntu-latest
+    outputs:
+      src: ${{ steps.changes.outputs.src }}
+    steps:
+    - uses: actions/checkout@v4
+    - uses: dorny/paths-filter@v2
+      id: changes
+      with:
+        filters: |
+          src:
+            - 'dockerfiles/requirements.txt'
+            - 'dockerfiles/Dockerfile.base'
```

Comment:
do we need to worried about any our dependency updates breaks CI? If I am get the points for line 17 and 18, which means any updates for dependency as xgboost version will skip unit test?
hence if any breaking changes happened in xgboost api, it will skipped CI and merged?

Reply:
I got the code, cool~